### PR TITLE
accesscontrol: add accesscontrol to apikey

### LINKED ISSUE #1547: Grafana User Survey  & Standalone mode question
I have created a user survey that I wish as many Grafana users could take the time to complete.  It will help a lot in knowing what the Grafana user community cares about (and how you use Grafana).

https://sv.surveymonkey.com/s/M93VR3G

So I am on the final stretch of getting Grafana 2.0 (with the new backend written in Go) ready for a beta release (hoping within 2 weeks). 

One issue that remains, which I am very unsure of is if I should spend time making Grafana 2.0 work in a standalone (i.e. no backend mode, exactly like Grafana v1.x, ie you have to setup nginx, and elasticsearch for dashboard storage). 

The standalone mode was working for a time, it took quite some effort to have a frontend architecture and asset pipeline (grunt build step) that could support both. But over the last month and a half the standalone mode has been a burden and in order to move faster with the backend features it is now pretty broken. 

So I want to ask the Grafana user community if the standalone mode is really something that is important to spend time on. Many of the new features coming in future versions will require the backend, some should not (like new panels).

So why does Grafana 2.0 has a backend? 
Some things the backend enable:
* Integrated user & dashboard storage in sqlite3 (or external in mysql or postgres) 
* Panel to PNG rendering (will enable really nice integration with Slack and Hipchjat)
* Integration with alerting (will come in future release, not in 2.0) 
* Nice UI for adding data sources  (no more config.js editing)
* HTTP api 
* Easy install & setup (the backend can proxy data source requests, no more nginx  and CORS setup issues)
* A Generic SQL metric data source
* + 100 of future improvements that I cannot think of right 

Benefits of a standalone mode: 
* Can run grafana only with nginx and apache no custom grafana backend binary needed

Again, please checkout the survey:
https://sv.surveymonkey.com/s/M93VR3G

### PR DESCRIPTION
**Pull request by donomii**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/1547

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -247,9 +247,9 @@ func (hs *HTTPServer) registerRoutes() {
 
 		// auth api keys
 		apiRoute.Group("/auth/keys", func(keysRoute routing.RouteRegister) {
-			keysRoute.Get("/", routing.Wrap(GetAPIKeys))
-			keysRoute.Post("/", quota("api_key"), bind(models.AddApiKeyCommand{}), routing.Wrap(hs.AddAPIKey))
-			keysRoute.Delete("/:id", routing.Wrap(DeleteAPIKey))
+			keysRoute.Get("/", authorize(reqOrgAdmin, accesscontrol.ActionApiKeyList, accesscontrol.ScopeUsersAll), routing.Wrap(GetAPIKeys))
+			keysRoute.Post("/", authorize(reqOrgAdmin, accesscontrol.ActionApiKeyAdd, accesscontrol.ScopeUsersAll), quota("api_key"), bind(models.AddApiKeyCommand{}), routing.Wrap(hs.AddAPIKey))
+			keysRoute.Delete("/:id", authorize(reqOrgAdmin, accesscontrol.ActionApiKeyDelete, accesscontrol.ScopeUsersAll), routing.Wrap(DeleteAPIKey))
 		}, reqOrgAdmin)
```
Comment (kalleep) [inline pkg/api/api.go:253]:
```suggestion
		})
```

File: pkg/api/api.go [hunk 2]
```diff
@@ -247,9 +247,9 @@ func (hs *HTTPServer) registerRoutes() {
 
 		// auth api keys
 		apiRoute.Group("/auth/keys", func(keysRoute routing.RouteRegister) {
-			keysRoute.Get("/", routing.Wrap(GetAPIKeys))
-			keysRoute.Post("/", quota("api_key"), bind(models.AddApiKeyCommand{}), routing.Wrap(hs.AddAPIKey))
-			keysRoute.Delete("/:id", routing.Wrap(DeleteAPIKey))
+			keysRoute.Get("/", authorize(reqOrgAdmin, accesscontrol.ActionApiKeyList, accesscontrol.ScopeUsersAll), routing.Wrap(GetAPIKeys))
```
Comment (kalleep) [inline pkg/api/api.go:250]:
```suggestion
			keysRoute.Get("/", authorize(reqOrgAdmin, accesscontrol.ActionApiKeyList), routing.Wrap(GetAPIKeys))
```

File: pkg/api/api.go [hunk 3]
```diff
@@ -247,9 +247,9 @@ func (hs *HTTPServer) registerRoutes() {
...
+			keysRoute.Post("/", authorize(reqOrgAdmin, accesscontrol.ActionApiKeyAdd, accesscontrol.ScopeUsersAll), quota("api_key"), bind(models.AddApiKeyCommand{}), routing.Wrap(hs.AddAPIKey))
```
Comment (kalleep) [inline pkg/api/api.go:251]:
```suggestion
			keysRoute.Post("/", authorize(reqOrgAdmin, accesscontrol.ActionApiKeyAdd), quota("api_key"), bind(models.AddApiKeyCommand{}), routing.Wrap(hs.AddAPIKey))
```

File: pkg/api/api.go [hunk 4]
```diff
@@ -247,9 +247,9 @@ func (hs *HTTPServer) registerRoutes() {
...
+			keysRoute.Delete("/:id", authorize(reqOrgAdmin, accesscontrol.ActionApiKeyDelete, accesscontrol.ScopeUsersAll), routing.Wrap(DeleteAPIKey))
```
Comment (kalleep) [inline pkg/api/api.go:252]:
```suggestion
			keysRoute.Delete("/:id", authorize(reqOrgAdmin, accesscontrol.ActionApiKeyDelete), routing.Wrap(DeleteAPIKey))
```

File: pkg/api/api.go (hunk 3 above)
Comment (kalleep) [reply pkg/api/api.go:251]: I dont thing you need to have scope for this

**Review by kalleep — COMMENTED** [review]
You need to add a new fixed role and grant that role to org admins

File: pkg/api/api.go (hunk 3 above)
Comment (donomii) [reply pkg/api/api.go:251]: So no scope is all scope?

**Comment by donomii** [comment]
Obsolete, redone in https://github.com/grafana/grafana/pull/42659

**Pull request closed without merging**